compiler-core/src/error.rs

```diff
@@ -3626,6 +3630,8 @@ or you can publish it using a different version number"),
                 location: None,
                 hint: Some("Please add the --replace flag if you want to replace the release.".into()),
             }],
+            Error::BuildVersionMismatch => vec![],
```

Comment:
I feel like these two not fit well to the rest of the values, because they are manually handled as a result of a `check_gleam_version` call. Is there a better way of doing it?